AirCasting/SearchAndFollow/CompleteScreen/Views/CompleteScreenViewModel.swift

```diff
@@ -141,6 +145,51 @@ class CompleteScreenViewModel: ObservableObject {
         exitRoute()
     }
     
+    private func refresh() {
+        guard ((session.stream.first?.sensorName.contains("AirBeam")) != nil) else { getMeasurementsAndDisplayData(); return }
+        var currentSensor = AirBeamStreamPrefix.airBeam3
+        if session.stream.first!.sensorName.contains("AirBeam2") { currentSensor = .airBeam2 }
+        do {
+            try streamsDownloader.downloadStreams(with: session.id, for: currentSensor) { result in
+                switch result {
+                case .success(let downloadedStreams):
+                    
+                    // TODO: - FIX Thresholds, get those values from backend not from our hardcoded struct
+                    #warning("🚨 FIX Thresholds 🚨")
+                    let streamHardcodedData = [MeasurementStream(sensorName: .f, sensorPackageName: ""),
+                                               MeasurementStream(sensorName: .pm1, sensorPackageName: ""),
+                                               MeasurementStream(sensorName: .pm10, sensorPackageName: ""),
+                                               MeasurementStream(sensorName: .pm2_5, sensorPackageName: ""),
+                                               MeasurementStream(sensorName: .rh, sensorPackageName: "")]
+                    
+                    let sessionStream = downloadedStreams.map({ stream -> PartialExternalSession.Stream in
```

Comment:
Thanks, this name is much better. I would change it to plural though, as it is holding multiple streams